sst/test/include/framework/registry.hh

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "framework/context.hh"
+#include "framework/runner.hh"
+#include "framework/uid.hh"
+
+/**
+ * The Registry class is used register Factory(s) which are able
+ * to create Runner(s) associated with TEST_GROUP(s). Registry is a singleton
+ * and can be only be instantiated once.
+ */
+class Registry {
+ public:
+  using Factory = std::function<std::unique_ptr<Runner>()>;
+  using RegistryMap = std::map<std::string, std::vector<Factory>>;
+
+  /** Method creates and returns the singleton instance of Registry. */
+  static Registry* getInstance() {
+    /** Static instance of the singleton Registry class. */
+    static Registry* instance_;
+    if (instance_ == NULL) {
+      instance_ = new Registry();
+    }
+    return instance_;
+  }
```

Comment:
What is the advantage of this function over a traditional constructor?

Reply:
Referring to the response left for this [comment](https://github.com/UoB-HPC/SimEng/pull/250#discussion_r991511710). 
Doing this always ensures we always have a singleton, without having to ensure that manually by passing references and ensuring nothing is copied, this way makes it more manageable and cleaner.